backend/routes/products.js

```diff
@@ -0,0 +1,46 @@
+const express = require("express");
+const router = express.Router();
+const Product = require("../models/Product");
+
+// return list of products
+router.get("/", async (req, res) => {
+  const productsData = await Product.find({});
+  res.send({ productsData });
+});
+//
+
+// tpost request for new products
+router.post("/add-new", async (req, res) => {
+  const title = req.body.title;
+  const image = req.body.image;
+  const description = req.body.description;
+  const category = req.body.category;
+  const price = req.body.price;
+
+  if (!title || !image || !description || !category || !price) {
+    res
+      .status(400)
+      .send({ message: "Please provide all the necessary details" });
+  }
+
+  const productExists = await Product.findOne({ title: title });
```

Comment:
Read about exists in mongo documentation and you might improve this line of code, or add .lean() in the end.

Reply:
const productExists = await Product.find({ title }).limit(1).lean();